--- icu.orig/source/common/putil.c	2007-12-12 19:57:26 +0100
+++ icu/source/common/putil.c	2008-01-20 21:08:57 +0100
@@ -105,7 +105,7 @@
 #   define ICU_NO_USER_DATA_OVERRIDE 1
 #elif defined(OS390)
 #include "unicode/ucnv.h"   /* Needed for UCNV_SWAP_LFNL_OPTION_STRING */
-#elif defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD)
+#elif defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD) || defined(U_KBSD)
 #include <limits.h>
 #include <unistd.h>
 #elif defined(U_QNX)
@@ -573,10 +573,10 @@
 extern U_IMPORT char *U_TZNAME[];
 #endif
 
-#if !UCONFIG_NO_FILE_IO && (defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD))
+#if !UCONFIG_NO_FILE_IO && (defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD) || defined(U_KBSD))
 /* These platforms are likely to use Olson timezone IDs. */
 #define CHECK_LOCALTIME_LINK 1
-#if defined(U_LINUX)
+#if defined(U_LINUX) || defined(U_KBSD)
 #define TZDEFAULT       "/etc/localtime"
 #define TZZONEINFO      "/usr/share/zoneinfo/"
 #else
@@ -1427,7 +1427,7 @@
         */
         name = "eucjis";
     }
-#elif defined(U_LINUX)
+#elif defined(U_LINUX) || defined(U_KBSD)
     if (locale != NULL && uprv_strcmp(name, "euc") == 0) {
         /* Linux underspecifies the "EUC" name. */
         if (uprv_strcmp(locale, "korean") == 0) {
--- icu.orig/source/configure	2007-12-12 19:58:06 +0100
+++ icu/source/configure	2008-01-20 21:07:42 +0100
@@ -8291,6 +8291,7 @@
 case "${host}" in
 	*-*-solaris*) 	platform=U_SOLARIS ;;
 	*-*-linux*) 	platform=U_LINUX ;;
+	*-*-k*bsd*-gnu*) platform=U_KBSD ;;
 	*-pc-gnu) 	platform=U_HURD ;;
 	*-*-*bsd*|*-*-dragonfly*) 	platform=U_BSD ;;
 	*-*-aix*) 	platform=U_AIX ;;
--- icu.orig/source/configure.in	2007-12-12 19:58:06 +0100
+++ icu/source/configure.in	2008-01-20 21:07:42 +0100
@@ -996,6 +996,7 @@
 case "${host}" in
 	*-*-solaris*) 	platform=U_SOLARIS ;;
 	*-*-linux*) 	platform=U_LINUX ;;
+	*-*-k*bsd*-gnu*) platform=U_KBSD ;;
 	*-pc-gnu) 	platform=U_HURD ;;
 	*-*-*bsd*|*-*-dragonfly*) 	platform=U_BSD ;;
 	*-*-aix*) 	platform=U_AIX ;;
