Index: icu/source/common/unicode/rbbi.h
===================================================================
--- icu.orig/source/common/unicode/rbbi.h	2009-09-07 19:53:29.017855637 -0400
+++ icu/source/common/unicode/rbbi.h	2009-09-07 19:53:57.165855852 -0400
@@ -611,12 +611,14 @@
     virtual int32_t getBreakType() const;
 #endif
 
+public:
     /**
       * Set the type of the break iterator.
       * @internal
       */
     virtual void setBreakType(int32_t type);
 
+protected:
     /**
       * Common initialization function, used by constructors and bufferClone.
       *   (Also used by DictionaryBasedBreakIterator::createBufferClone().)
