Description: changes for Debian kfreebsd port
Author: Aurelien Jarno <aurel32@debian.org>
Bug-Debian: http://bugs.debian.org/461782
Bug: http://bugs.icu-project.org/trac/ticket/6406

Index: icu-4.3.4/source/common/putil.c
===================================================================
--- icu-4.3.4.orig/source/common/putil.c	2010-02-06 17:01:01.703257361 -0500
+++ icu-4.3.4/source/common/putil.c	2010-02-06 17:01:04.419258219 -0500
@@ -106,7 +106,7 @@
 #   define ICU_NO_USER_DATA_OVERRIDE 1
 #elif defined(OS390)
 #include "unicode/ucnv.h"   /* Needed for UCNV_SWAP_LFNL_OPTION_STRING */
-#elif defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD)
+#elif defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD) || defined(U_KBSD)
 #include <limits.h>
 #include <unistd.h>
 #elif defined(U_QNX)
@@ -631,7 +631,7 @@
 extern U_IMPORT char *U_TZNAME[];
 #endif
 
-#if !UCONFIG_NO_FILE_IO && (defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD))
+#if !UCONFIG_NO_FILE_IO && (defined(U_DARWIN) || defined(U_LINUX) || defined(U_BSD) || defined(U_KBSD))
 /* These platforms are likely to use Olson timezone IDs. */
 #define CHECK_LOCALTIME_LINK 1
 #if defined(U_DARWIN)
@@ -1702,7 +1702,7 @@
         */
         name = "eucjis";
     }
-#elif defined(U_LINUX)
+#elif defined(U_LINUX) || defined(U_KBSD)
     if (locale != NULL && uprv_strcmp(name, "euc") == 0) {
         /* Linux underspecifies the "EUC" name. */
         if (uprv_strcmp(locale, "korean") == 0) {
Index: icu-4.3.4/source/configure
===================================================================
--- icu-4.3.4.orig/source/configure	2010-02-06 17:01:01.715257667 -0500
+++ icu-4.3.4/source/configure	2010-02-06 17:01:04.423258176 -0500
@@ -11390,6 +11390,7 @@
 case "${host}" in
 	*-*-solaris*) 	platform=U_SOLARIS ;;
 	*-*-linux*) 	platform=U_LINUX ;;
+	*-*-k*bsd*-gnu*) platform=U_KBSD ;;
 	*-pc-gnu) 	platform=U_HURD ;;
 	*-*-*bsd*|*-*-dragonfly*) 	platform=U_BSD ;;
 	*-*-aix*) 	platform=U_AIX ;;
Index: icu-4.3.4/source/configure.in
===================================================================
--- icu-4.3.4.orig/source/configure.in	2010-02-06 17:01:01.723257837 -0500
+++ icu-4.3.4/source/configure.in	2010-02-06 17:01:04.431257137 -0500
@@ -1197,6 +1197,7 @@
 case "${host}" in
 	*-*-solaris*) 	platform=U_SOLARIS ;;
 	*-*-linux*) 	platform=U_LINUX ;;
+	*-*-k*bsd*-gnu*) platform=U_KBSD ;;
 	*-pc-gnu) 	platform=U_HURD ;;
 	*-*-*bsd*|*-*-dragonfly*) 	platform=U_BSD ;;
 	*-*-aix*) 	platform=U_AIX ;;
