Description: work around incorrect assembly generation in ICU build
Author: Jay Berkenbilt <qjb@debian.org>
Bug: http://bugs.icu-project.org/trac/ticket/7119

Index: icu-4.3.4/source/tools/pkgdata/pkgdata.cpp
===================================================================
--- icu-4.3.4.orig/source/tools/pkgdata/pkgdata.cpp	2010-02-06 17:01:01.619257367 -0500
+++ icu-4.3.4/source/tools/pkgdata/pkgdata.cpp	2010-02-06 17:01:04.935257423 -0500
@@ -77,6 +77,13 @@
 #define CAN_WRITE_OBJ_CODE
 #endif
 
+// Temporary fix until http://bugs.icu-project.org/trac/ticket/7119 is
+// resolved.
+#ifdef CAN_WRITE_OBJ_CODE
+# undef CAN_WRITE_OBJ_CODE
+# define BUILD_DATA_WITHOUT_ASSEMBLY
+#endif
+
 /*
  * When building the data library without assembly,
  * some platforms use a single c code file for all of
